src/sql/workbench/contrib/notebook/browser/models/notebookInput.ts

```diff
@@ -248,6 +248,9 @@ export abstract class NotebookInput extends EditorInput implements INotebookInpu
 		this._providersLoaded = this.assignProviders();
 		this._notebookEditorOpenedTimestamp = Date.now();
 		if (this._textInput) {
+			if (this._textInput instanceof UntitledTextEditorInput) {
+				this._textInput.setMode('notebook');
```

Comment:
Is there any reason why we wouldn't just always set the mode to Notebook? It's probably unnecessary for non-untitled files but it shouldn't harm anything and will make this more robust.

Reply:
I agree and I was trying to do that but setMode is not implemented by one of the editor inputs - TextResourceEditorModel. So I was not able to do it directly.

Reply:
Ah, then you should just put this in https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/models/untitledNotebookInput.ts

Reply:
You're sure this works for you? I wanted to test if this fixed something else but : 

1. In the constructor the mode is already set to 'notebook' for the text input
2. Regardless of that, even with the code to set the mode I'm not seeing any difference. The save dialog still prompts to save as JSON

![image](https://user-images.githubusercontent.com/28519865/149601127-b6bb874f-294f-43ef-9d59-f731510d40e6.png)

Reply:
@Charles-Gagnon It's the same in the release version as well, so I guess we always save as type JSON but in the name the extension is ipynb.